gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/HiveProxyQueryExecutor.java

```diff
@@ -0,0 +1,223 @@
+          ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_DEFAULT_SHOULD_PROXY)) {
+        setConnection();
+        return;
+      }
+      Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_PROXY_USER),
+          "Missing required property " + ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_PROXY_USER);
+      Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_SUPER_USER),
+          "Missing required property " + ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_SUPER_USER);
+      List<String> proxies = new ArrayList<>();
+      proxies.add(this.state.getProp(ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_PROXY_USER));
+      proxies.add(this.state.getProp(ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_SUPER_USER));
+      setProxiedConnection(proxies);
+    } catch (ClassNotFoundException | SQLException | InterruptedException | TException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private synchronized void setProxiedConnection(final List<String> proxies)
+      throws IOException, InterruptedException, TException {
+    Preconditions.checkArgument(this.state.contains(ConfigurationKeys.SUPER_USER_KEY_TAB_LOCATION),
+        "Missing required property " + ConfigurationKeys.SUPER_USER_KEY_TAB_LOCATION);
+    String superUser = this.state.getProp(ComplianceConfigurationKeys.GOBBLIN_COMPLIANCE_SUPER_USER);
+    String keytabLocation = this.state.getProp(ConfigurationKeys.SUPER_USER_KEY_TAB_LOCATION);
+    String realm = this.state.getProp(ConfigurationKeys.KERBEROS_REALM);
+    UserGroupInformation loginUser = UserGroupInformation
+        .loginUserFromKeytabAndReturnUGI(HostUtils.getPrincipalUsingHostname(superUser, realm), keytabLocation);
+    loginUser.doAs(new PrivilegedExceptionAction<Void>() {
+      @Override
+      public Void run()
+          throws MetaException, SQLException, ClassNotFoundException {
+        for (String proxy : proxies) {
+          HiveConnection hiveConnection = getHiveConnection(proxy);
+          Statement statement = hiveConnection.createStatement();
+          statementMap.put(proxy, statement);
+          connectionMap.put(proxy, hiveConnection);
+          for (String setting : settings) {
+            statement.execute(setting);
+          }
+        }
+        return null;
+      }
+    });
+  }
+
+  private synchronized void setConnection()
+      throws ClassNotFoundException, SQLException {
+    HiveConnection hiveConnection = getHiveConnection();
+    Statement statement = hiveConnection.createStatement();
+    this.statementMap.put(DEFAULT, statement);
+    this.connectionMap.put(DEFAULT, hiveConnection);
+    for (String setting : settings) {
+      statement.execute(setting);
+    }
+  }
+
+  private HiveConnection getHiveConnection(String proxyUser)
+      throws ClassNotFoundException, SQLException {
+    Class.forName("org.apache.hive.jdbc.HiveDriver");
```

Comment:
Is this just to check for `ClassNotFoundException`? Won't DriverManager.getConnection() raise an error if it can't find an appropriate driver?